api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1923,4 +1923,23 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
 	
+	/**
+	 * Get the concept by conceptRef where the conceptRef can either be: 1) an integer id like 5090
+	 * 2) a mapping type id like "XYZ:HT" 3) a uuid like "a3e12268-74bf-11df-9768-17cfc9833272" 4) a
+	 * name like "PLATELETS" 5) the fully qualified name of a Java constant that contains one of
+	 * above
+	 * 
+	 * @param conceptRef the concept string identifier
+	 * @return the concept if exist, else null
+	 * <strong>Should</strong> find a concept by its conceptId
+	 * <strong>Should</strong>  find a concept by its mapping
+	 * <strong>Should</strong> find a concept by its uuid
+	 * <strong>Should</strong> find a concept by its name
+	 * <strong>Should</strong> find a concept by a fully qualified name of a Java constant that contains an id,
+	 * 								uuid, name or mapping
+	 * <strong>Should</strong> return null when ConceptRef does not exist
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public Concept getConceptByNameOrIdOrMap(String conceptRef);
```

Comment:
I'd call this `getConceptByReference()`, This should have a `@since` comment in the Javadoc. Also, we don't need the `<should>` stuff, although I would expect these to be included in unit tests.

Reply:
Thank you @ibacher 
> although I would expect these to be included in unit tests.

The way i am understanding this is that i should remove the **<strong>Should</strong>** statements but persist the corresponding unit tests.

Reply:
yes, the corresponding unit tests should be maintained